docs/howto/customization/workflows.md

```diff
@@ -1,183 +1,559 @@
+    - name: Input system
+      section: '../upload/archive/mainfile/dft.xml#/run/0/system/-1'
   outputs:
-    - name: relaxed system
-      section: '../upload/raw/geom_opt.archive.yaml#/run/0/system/-1'
-    - name: ground state calculation of relaxed system
-      section: '../upload/raw/ground_state.archive.yaml#/run/0/calculations/0'
+    - name: Output calculation
+      section: '../upload/archive/mainfile/dft.xml#/run/0/calculation/-1'
   tasks:
-    - name: GeometryOpt
+    - m_def: nomad.datamodel.metainfo.workflow.TaskReference
+      task: '../upload/archive/mainfile/dft.xml#/workflow2'
+      name: DFT
       inputs:
-        - name: input system
-          section: '../upload/raw/geom_opt.archive.yaml#/run/0/system/0'
+        - name: Input structure
+          section: '../upload/archive/mainfile/dft.xml#/run/0/system/-1'
       outputs:
-        - name: relaxed system
-          section: '../upload/raw/geom_opt.archive.yaml#/run/0/system/-1'
+        - name: Output calculation
+          section: '../upload/archive/mainfile/dft.xml#/run/0/calculation/-1'
+```
 
-    - name: GroundStateCalculation
-      inputs:
-        - name: input system
-          section: '../upload/raw/geom_opt.archive.yaml#/run/0/system/-1'
-      outputs:
-        - name: ground state
-          section: '../upload/raw/ground_state.archive.yaml#/run/0/calculations/0'
+!!! Warning "Important"
+    For the creation of workflow entries using YAMLs, the file must have the extension `archive.yaml`.
+
+This file is constructed according to NOMAD's [General Workflow Schema](../../explanation/workflows.md#the-built-in-abstract-workflow-schema). The `workflow2` section of the archive has 3 possible subsections: `inputs`, `outputs`, and `tasks`:
+
+**`inputs`**: a list of references to the global inputs of the workflow, with `name` and `section` attributes. `section` corresponds to a path for linking to the relevant archive section. In this case, the relative section path is `run[0].system[-1]`, linked to the entry defined by the mainfile `dft.xml`. The prefix is discussed under "Considerations for archive path specification" directly below.
+
+**`outputs`**: identical to the inputs list, representing the global outputs of the workflow, with the relative section path `run[0].calculation[-1]` in this case.
+
+**`tasks`**: a list of references to the tasks/steps of the workflow. Each task contains `m_def`, `task`, `inputs`, and `outputs` attributes. `inputs`/`outputs` are task-specific versions of the lists defined above. `task` is the path for linking to the relevant archive section, analogous to the `section` attribute for `inputs`/`outputs`. `m_def` defines the type of task according to NOMAD's MetaInfo scehma, in this case a `TaskReference` to the archive `workflow2` section.  The use of `TaskReference` will be clarified in the [Nested workflows > In multiple entries](#in-multiple-entries) example below.
```

Comment:
'task is the path for linking to the relevant archive section', but it must only be a reference to a task which is practically only a `workflow2` section